lib/ui/painting.dart

```diff
@@ -323,13 +423,21 @@ class Color {
     if (other.runtimeType != runtimeType) {
       return false;
     }
-    return other is Color
-        && other.value == value;
+    // TODO(gaaclarke): Make equality more fine grain than 1/256.
```

Comment:
You should change the `==` implementation in this PR and not in a follow up. You'll want some sort of fudge factor for the comparisons.

Reply:
Thinking about this some more, we want ensure the hashCode is equal if the color is equal, so normaliziing to some bit depth (not 256) is probably required here.

Reply:
Okay, so I had this all implemented and decided to pull it out and wait for new PR.  The reason why is that I want to separate the PRs for landing support for rendering wide gamut colors and support for wider bit depth for Color components.  The reason is that the wider bit depth in color is the thing that is most likely going to cause a regression in tests.  Rendering wide gamut colors is 3 commits across 2 repos and will be a pain to revert.  I really think these should be landed separately.

My plan is to basically do the same math but use a number like 2^12 to match the bit depth for HDMI 2.0.  On iOS we currently can only render 2^10.

Reply:
We could switch the storage back to an int in the meantime if it makes landing this PR easier.

Reply:
In fact maybe it makes sense to switch the storage to always be fixed point.  I could refactor the old code to work from a fixed point where the divisor is 255.  Then we can crank it up to 6000.  The transmission to the lower levels can remain floats.  The benefit is that the change at dart:ui is much simpler.  Having 6000 be the fixed-point divisor would allow us to accurately represent 1/2, 1/4, 1/8, 1/10, 1/16.  It's close to the 2^12 bit rate.